### PR TITLE
Protractor tests can failed Travis build

### PR DESCRIPTION
**Pull request by pascalgrimaud**
I add 2 news tasks in gulpfile.js : itest-travis and protractor-travis

So when the protractor tests failed, it should return 1 to travis build, so it should failed now...
I add protractor tests to MySQL sample in production profile.
So I had to remove the test on "api docs" because it doesn't exist in prod profile

### CODE REVIEW

File: generators/client/templates/gulpfile.js [hunk 1]
```diff
@@ -53,14 +53,32 @@ gulp.task('test', ['wiredep:test', 'ngconstant:dev'], function (done) {
...
+});
+
+gulp.task('protractor-travis', function () {
+    gulp.src([config.test + 'e2e/**/*.js'])
+        .pipe(protractor({
+            configFile: config.test + 'protractor.conf.js'
+        }))
+        .on('end', function() {
+            console.log('E2E Testing complete');
+            // exit with success.
```
Comment (deepu105) [inline generators/client/templates/gulpfile.js]: on success the process seems to be exiting already so we might not need this

File: generators/client/templates/gulpfile.js [hunk 2]
```diff
@@ -53,14 +53,32 @@ gulp.task('test', ['wiredep:test', 'ngconstant:dev'], function (done) {
...
+            process.exit(0);
+        })
+        .on('error', function(err) {
+            console.log('E2E Tests failed');
+            process.exit(1);
```
Comment (deepu105) [inline generators/client/templates/gulpfile.js]: this can be added to the actual itest task as its ok to exit when there is error.

**Comment by deepu105** [comment]
Im not ok with adding a task to user file which is only used for our travis build better just amend the protractor task like this

```
gulp.task('protractor', function () {
    return gulp.src([config.test + 'e2e/**/*.js'])
        .pipe(plumber({errorHandler: handleErrors}))
        .pipe(protractor({
            configFile: config.test + 'protractor.conf.js'
        }))
        .on('error', function(err) {
              console.log('E2E Tests failed');
              process.exit(1);
         });
});
```

**Comment by deepu105** [comment]
also may be we should enable protractor for gradle build as well?? what do you think

**Comment by pascalgrimaud** [comment]
Yeah it was late in France, bad thinking. Thanks for review @deepu105 

About Gradle build, there is just `gradlew test` and `gulp test`, because there is no cache and the build is long. So the application isn't started. If we add protractor tests for gradle :
1) start app -> 1~2min
2) protractor -> 1min
Is it worth ?

**Comment by deepu105** [comment]
anyway our travis takes atleast 30 mins normally to fully finish so I dont
think this will increase that as other builds would be running parallel. So
I think it would be ok to add

Thanks & Regards,
Deepu

**Comment by pascalgrimaud** [comment]
For gradle, I will do it in a separate PR

File: generators/client/templates/src/test/javascript/e2e/admin/_administration.js [hunk 3]
```diff
@@ -55,11 +55,6 @@ describe('administration', function () {
         expect(element.all(by.css('h2')).first().getText()).toMatch(/Logs/);
     });
 
-    it('should load api docs', function () {
```
Comment (deepu105) [inline generators/client/templates/src/test/javascript/e2e/admin/_administration.js]: why remove this?

**Comment by pascalgrimaud** [comment]
1) it's for testing in prod profile => mysql sample
there is no api docs in prod profile
I could put it back but we won't be able to test in prod profile, like suggested by @gmarziou in our ML

2) the current protractor tests passed and failed randomly and I don't understand why. In my local, it passed the 1st time, but if i run again `gulp ites`t, I'm stuck at login page => @deepu105 can you run `gulp itest` twice ?

**Comment by deepu105** [comment]
Ok api docs are not required in test i guess.

I was able to run itest twice yesterday ill try again today

Thanks & Regards,
Deepu

**Comment by deepu105** [comment]
I rerun but Protractor tests are again failing

**Comment by pascalgrimaud** [comment]
Yes, my modifications doesn't modify the protractor tests, just reveal the truth :alien: 
What I don't understand is :
- on my local, it failed only the 2nd time, the 1st time is ok
- on Travis, it failed immediatly
- on your local, @deepu105, it passed ?

**Comment by deepu105** [comment]
I had it passing without your changes, ill try your branch today

**Comment by deepu105** [comment]
but protractor is definitely failing in master as well. I guess It might have something to do with the gulp-protractor upgrade let me try in local

**Comment by deepu105** [comment]
I tried in my laptop and protractor tests are working fine for current master as well for your branch. And I have no issues when running two tests back to back

But Im on windows

```

Deepu@DESKTOP-3NUQHBI  /d/Projects/test/testApp
$ gulp itest
[18:52:38] Using gulpfile D:\Projects\test\testApp\gulpfile.js
[18:52:38] Starting 'protractor'...
Using FirefoxDriver directly...
[launcher] Running 1 instances of WebDriver
Started
..........


10 specs, 0 failures
Finished in 40.79 seconds
[launcher] 0 instance(s) of WebDriver still running
[launcher] firefox #1 passed
[18:53:30] Finished 'protractor' after 52 s
[18:53:30] Starting 'itest'...
[18:53:30] Finished 'itest' after 32 μs

Deepu@DESKTOP-3NUQHBI  /d/Projects/test/testApp
$ gulp itest
[18:53:37] Using gulpfile D:\Projects\test\testApp\gulpfile.js
[18:53:37] Starting 'protractor'...
Using FirefoxDriver directly...
[launcher] Running 1 instances of WebDriver
Started
..........


10 specs, 0 failures
Finished in 41.005 seconds
[launcher] 0 instance(s) of WebDriver still running
[launcher] firefox #1 passed
[18:54:28] Finished 'protractor' after 51 s
[18:54:28] Starting 'itest'...
[18:54:28] Finished 'itest' after 19 μs

Deepu@DESKTOP-3NUQHBI  /d/Projects/test/testApp
```

**Comment by pascalgrimaud** [comment]
Can you try manually :
1) go localhost:8080/
2) sign in
3) put admin/123, submit
4) put admin/admin, submit
5) log out
6) sign in
7) put admin/123, submit => is there an error on console like me ?
```
TypeError: Cannot read property 'split' of undefined
    at getCSRF (auth-expired.interceptor.js:40)
    at responseError (auth-expired.interceptor.js:30)
    at processQueue (angular.js:15552)
    at angular.js:15568
    at Scope.$eval (angular.js:16820)
    at Scope.$digest (angular.js:16636)
    at Scope.$apply (angular.js:16928)
    at done (angular.js:11266)
    at completeRequest (angular.js:11464)
    at XMLHttpRequest.requestLoaded (angular.js:11405)
```

Maybe come from: https://github.com/jhipster/generator-jhipster/blob/master/generators/client/templates/src/main/webapp/app/blocks/interceptor/_auth-expired.interceptor.js#L62
Should it be `document.cookie.split` instead ?

**Comment by pascalgrimaud** [comment]
With Windows7, protractor tests worked fine, there is no error with Firefox
But same error described above with Chrome

**Comment by deepu105** [comment]
I was running firefox as well ill test what you said in chrome today

**Comment by deepu105** [comment]
@pascalgrimaud may be first lets add the `.on('error'` block and later we can enable protractor for mysql prod build

**Comment by deepu105** [comment]
you are absolutely correct, im fixing this right away

**Comment by deepu105** [comment]
@pascalgrimaud sorry i stole your on error blvok, can you amend this PR to just add protractor for the myql prod build and to remove api doc test?

https://github.com/jhipster/generator-jhipster/commit/28a2b219a9d0625d5d4cfb632b9c13624779b582

**Comment by deepu105** [comment]
seems [this ](https://github.com/jhipster/generator-jhipster/commit/a490574d1762d7fac8bbb4ce82d3ee3cba75f159)was an issue as well

**Comment by pascalgrimaud** [comment]
Yes no pb man :-) we finally find the problem
I will rebase and update this PR soon

**Comment by deepu105** [comment]
finally protractor is passing :)

**Comment by deepu105** [comment]
may be you could add protractor for gradle also here

**Comment by pascalgrimaud** [comment]
I will close this PR and make another cleaner, with less useless commits from me!
Thanks for your help Deepu !

**Pull request closed without merging**